src/components/common/modals/common-modal.tsx

```diff
@@ -0,0 +1,34 @@
+import React, { Fragment } from 'react'
+import { Modal } from 'react-bootstrap'
+import { Trans } from 'react-i18next'
+import { ForkAwesomeIcon, IconName } from '../fork-awesome/fork-awesome-icon'
+import { ShowIf } from '../show-if/show-if'
+
+export interface CommonModalProps {
+  show: boolean
+  onHide: () => void
+  title: string
+  closeButton?: boolean
+  icon?: IconName
+}
+
+export const CommonModal: React.FC<CommonModalProps> = ({ show, onHide, title, closeButton, icon, children }) => {
+  return (
+    <Modal show={show} onHide={onHide} animation={true} className="text-dark">
+      <Modal.Header closeButton={!!closeButton}>
+        <Modal.Title>
+          <ShowIf condition={!!icon}>
+            <Fragment>
+              {/* eslint-disable-next-line @typescript-eslint/no-non-null-assertion */}
+              <ForkAwesomeIcon icon={icon!}/>&nbsp;<Trans i18nKey={title}/>
```

Comment:
```suggestion
             {icon ? <ForkAwesomeIcon icon={icon}/> : ''}&nbsp;<Trans i18nKey={title}/>
```
I know we want to avoid ternary operator in favor of ShowIf, but if that does not work, I like this better than the eslint disable comment

Reply:
You can use `as IconProp` to convince the compiler that you're sure that you have a value

Reply:
@mrdrogdrog That sounds like the best alternative.